File: submissions/igyuk-meg-a-compareto-t/2.py
def f(s):
 k = '`1234567890-=~!@#$%^&*()_+``qwertyuiop[]\QWERTYUIOP{}|`asdfghjkl;\'ASDFGHJKL:"`````zxcvbnm,./ZXCVBNM<>?'
 m = [[],[],[],[]]
 c = [set(), set(),set(),set()]
 for x in s:
  i = k.index(x)//27
  if x not in c[i]:
   c[i].add(x)
   m[i].append(x)
 [y.sort(key=lambda x: k.index(x)) for y in m]
 m.sort(key=lambda x: len(x) if len(x) > 0 else 99)
 print(''.join(m[0]))

File: submissions/igyuk-meg-a-compareto-t/2.rb
def f(s)
d="`1234567890-=~!@\#$%^&*()_+qwertyuiop[]\\QWERTYUIOP{}|asdfghjkl;'ASDFGHJKL:\"````zxcvbnm,./ZXCVBNM<>?"
puts s.split('').group_by{|x|(d.index x)/26}.to_a.map{|x|[x[1].uniq.sort_by{|x|d.index x}.join,x[1].count,x[1].uniq.count,x[0]]}.sort_by{|x|x.drop 1}[0][0]
end